File: Sesion 11/Marco teorico/2INTENTO (TRY)/2.cpp
#include <iostream>
#include <exception>
using namespace std;


int main() {
    int numerador, denominador, resultado;


    cout << "Ingrese el numerador: ";
    cin >> numerador;
    cout << "Ingrese el denominador: ";
    cin >> denominador;


    try {
        // Intentamos realizar la división
        if (denominador == 0) {
            throw runtime_error("Error: División por cero.");
        }
        resultado = numerador / denominador;
        cout << "El resultado es: " << resultado << endl;
    }
    catch (const exception& e) {
        // Capturamos la excepción y mostramos el mensaje de error
        cerr << "Excepción capturada: " << e.what() << endl;
    }


    return 0;
}

File: Sesion 11/Marco teorico/2INTENTO (TRY)/2.py
class DivisionByZeroError(Exception):
    def __init__(self, message):
        self.message = message
        super().__init__(self.message)

try:
    numerador = int(input("Ingrese el numerador: "))
    denominador = int(input("Ingrese el denominador: "))

    if denominador == 0:
        raise DivisionByZeroError("Error: División por cero.")

    resultado = numerador // denominador
    print("El resultado es:", resultado)

except DivisionByZeroError as e:
    print("Excepción capturada:", e)

except ValueError:
    print("Error: Ingrese un número entero válido.")
